AppHandling/Run-AlPipeline.ps1

```diff
@@ -1746,31 +1768,47 @@ Write-Host -ForegroundColor Yellow @'
         }
     }
 
+    $compilationParams = $Parameters + $CopParameters
+
+    # Run pre-compile script if specified
+    if($PreCompileApp) {
+        Write-Host "Running custom pre-compilation script..."
+
+        Invoke-Command -ScriptBlock $PreCompileApp -ArgumentList ([ref] $compilationParams)
+    }
+
     try {
-        Write-Host "`nCompiling $($Parameters.appProjectFolder)"
-        $withCops = $Parameters+$CopParameters
+        Write-Host "`nCompiling $($compilationParams.appProjectFolder)"
         if ($useCompilerFolder) {
-            $appFile = Compile-AppWithBcCompilerFolder @parameters
+            $appFile = Compile-AppWithBcCompilerFolder @compilationParams
         }
         else {
-            $appFile = Invoke-Command -ScriptBlock $CompileAppInBcContainer -ArgumentList $withCops
+            $appFile = Invoke-Command -ScriptBlock $CompileAppInBcContainer -ArgumentList $compilationParams
         }
     }
     catch {
         if ($escapeFromCops) {
             Write-Host "Retrying without Cops"
+            $compilationParams = $Parameters
             if ($useCompilerFolder) {
-                $appFile = Compile-AppWithBcCompilerFolder @parameters
+                $appFile = Compile-AppWithBcCompilerFolder @compilationParams
             }
             else {
-                $appFile = Invoke-Command -ScriptBlock $CompileAppInBcContainer -ArgumentList $Parameters
+                $appFile = Invoke-Command -ScriptBlock $CompileAppInBcContainer -ArgumentList $compilationParams
             }
         }
         else {
             throw $_
         }
     }
 
+    # Run post-compile script if specified
+    if($PostCompileApp) {
+        Write-Host "Running custom post-compilation script..."
+
+        Invoke-Command -ScriptBlock $PostCompileApp -ArgumentList $appFile, $compilationParams

```

Comment:
What does the param block in the post and precompile app look like?
Is that a HashTable like the other overrides ?